src/schemes/boundary/dummy_particles/dummy_particles.jl

```diff
@@ -378,75 +414,115 @@ function compute_pressure!(boundary_model, ::Union{PressureMirroring, PressureZe
+                                          distance, viscosity, cache, pressure,
+                                          pressure_offset)
     density_neighbor = particle_density(v_neighbor_system, neighbor_system, neighbor)
 
-    resulting_acc = neighbor_system.acceleration -
-                    current_acceleration(system, particle)
+    resulting_acceleration = neighbor_system.acceleration -
+                             current_acceleration(system, particle)
 
     kernel_weight = smoothing_kernel(boundary_model, distance)
 
-    pressure[particle] += (particle_pressure(v_neighbor_system, neighbor_system,
-                                             neighbor) +
-                           dot(resulting_acc, density_neighbor * pos_diff)) *
-                          kernel_weight
+    pressure[particle] += (pressure_offset
+                           +
+                           particle_pressure(v_neighbor_system, neighbor_system,
+                                             neighbor)
+                           +
+                           dynamic_pressure(boundary_density_calculator, density_neighbor,
+                                            v, v_neighbor_system, particle, neighbor,
+                                            system, neighbor_system)
+                           +
+                           dot(resulting_acceleration,
+                               density_neighbor * pos_diff)) * kernel_weight
 
     cache.volume[particle] += kernel_weight
 
     compute_smoothed_velocity!(cache, viscosity, neighbor_system, v_neighbor_system,
                                kernel_weight, particle, neighbor)
 end
 
+@inline function dynamic_pressure(boundary_density_calculator, density_neighbor, v,
+                                  v_neighbor_system, particle,
+                                  neighbor, system, neighbor_system)
+    return zero(density_neighbor)
+end
+
+@inline function dynamic_pressure(::BernoulliPressureExtrapolation, density_neighbor, v,
+                                  v_neighbor_system, particle,
+                                  neighbor, system::BoundarySystem, neighbor_system)
+    if system.ismoving[]
+        relative_velocity = current_velocity(v, system, particle) .-
+                            current_velocity(v_neighbor_system, neighbor_system, neighbor)
+        return density_neighbor * dot(relative_velocity, relative_velocity) / 2
+    end
+    return zero(density_neighbor)
+end
+
+@inline function dynamic_pressure(::BernoulliPressureExtrapolation, density_neighbor, v,
+                                  v_neighbor_system, particle,
+                                  neighbor, system::SolidSystem, neighbor_system)
+    relative_velocity = current_velocity(v, system, particle) .-
+                        current_velocity(v_neighbor_system, neighbor_system, neighbor)
+    return density_neighbor * dot(relative_velocity, relative_velocity) / 2
+end
```

Comment:
This will increase the dynamic pressure even if the relative velocity is tangential to the boundary. The pressure must not increase for shear motion. It should only consider the projection normal to the boundary.

Reply:
The main objective of this boundary pressure calculation is to prevent penetration of particles like it happened in #496 and #522.
In general the necessary surface normal of a solid body is not available this can be fixed when used by defining a surface_normal_method as defined in #539 which can than be used here but the overhead of the surface normal calculation which is around 30% is not necessary in all cases.